src/prefabs/pageWithCreateDataTable.js

```diff
@@ -0,0 +1,12336 @@
+(() => ({
+  name: 'Page With Create Data Table',
+  description: 'This page contains a datatable with create dialog',
+  type: 'page',
+  icon: 'DataTable',
+  category: 'DATA',
+  beforeCreate: ({
+    helpers: { useModelQuery, camelToSnakeCase },
+    components: {
+      Header,
+      Content,
+      Footer,
+      Field,
+      Text,
+      CheckBox,
+      ModelSelector,
+      PropertiesSelector,
+    },
+    prefab,
+    save,
+    close,
+  }) => {
+    const [modelId, setModelId] = React.useState('');
+    const [properties, setProperties] = React.useState([]);
+    const [modelValidation, setModelValidation] = React.useState(false);
+    const [propertiesValidation, setPropertiesValidation] = React.useState(
+      false,
+    );
+    const [
+      createPropertiesValidation,
+      setCreatePropertiesValidation,
+    ] = React.useState(false);
+    const [createFormProperties, setCreateFormProperties] = React.useState([]);
+    const [
+      createFormUseDataProperties,
+      setCreateFormUseDataProperties,
+    ] = React.useState(true);
+
+    const { data } = useModelQuery({
+      variables: { id: modelId },
+      skip: !modelId,
+    });
+
+    const reduceStructure = (refValue, structure) =>
```

Comment:
This is just a suggestion. it's not wrong like it is right now, see the other PR https://github.com/bettyblocks/material-ui-component-set/pull/1048. But it looks like a lookup for a certain prefab in the structure, though the function name is rather global. Maybe you can think of something more descriptive.